impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -159,10 +160,17 @@ public TestcontainersKafkaCluster(KafkaClusterConfig clusterConfig) {
      * @param clusterConfig  the cluster config
      */
     public TestcontainersKafkaCluster(DockerImageName kafkaImage, DockerImageName zookeeperImage, KafkaClusterConfig clusterConfig) {
-        setDefaultKafkaImage(clusterConfig.getKafkaVersion());
+        var tag = getContainerImageVersionTag(clusterConfig.getKafkaVersion());
+
+        var actualKafkaImage = Optional.ofNullable(kafkaImage).orElse(DEFAULT_KAFKA_IMAGE);
+        var actualZookeeperImage = Optional.ofNullable(zookeeperImage).orElse(DEFAULT_ZOOKEEPER_IMAGE);
+        if (tag != null) {
+            actualKafkaImage = actualKafkaImage.withTag(tag);
+            actualZookeeperImage = actualZookeeperImage.withTag(tag);
```

Comment:
It's slightly weird that the zookeeper native images are tagged with the kafka version, but that's really a concern for elsewhere.  Zookeeper will wither away anyway 🥀.